web-commons/src/main/resources/templates/welcome.html

```diff
@@ -55,6 +58,122 @@
                 </ul>
             </div>
         </nav>
+        <script type="text/javascript">
+            function withDomainColumn(columns, currentColumn, domainColumn) {
+                if (currentColumn === domainColumn) {
+                    return {
+                        sourceColumn: currentColumn,
+                        label: columns[currentColumn],
+                        role: 'domain'
+                    }
+                } else {
+                    return {
+                        sourceColumn: currentColumn,
+                        label: columns[currentColumn],
+                        role: 'data'
+                    }
+                }
+            }
+
+            function pickDomainColumn(columns, domainColumnIndex) {
+                const allColumnsIndices = [...Array(columns.length).keys()]
+                return allColumnsIndices.map(i => withDomainColumn(columns, i, domainColumnIndex))
+            }
+
+            function commonHighChartsPieChartsOptions() {
```

Comment:
There is no point to make this a function I think. What about extracting common parts to make it more clear where they differ?